### ISSUE TITLE
Exporting destructured assignment causes no-undef

### ISSUE THREAD

**Issue by shipstar** [babel/babel-eslint#57]
This is a bit of an edge case, but I'm currently dealing with some variables that are on the window that I want to wrap up for browserify. I created a WindowService like the following:

    /*global window*/

    export default {
      foo,
      bar,
      baz
    } = window

but I'm getting the following errors when I lint:

     app/services/window.js
      4:6  error  'foo' is not defined  no-undef
      5:6  error  'bar' is not defined  no-undef
      6:6  error  'baz' is not defined  no-undef

Seems similar to but distinct from #56.

**Comment by sebmck** [comment]
Can you please verify that this issue still exists on 3.0.1?

**Comment by shipstar** [comment]
Still seems to be happening. Upgraded to eslint 0.19.0 and babel-eslint 3.0.1.

**Comment by sebmck** [comment]
Actually no this is expected behaviour. You're assigning to `foo`, `bar` and `baz` and they aren't defined.

**Comment by shipstar** [comment]
The following doesn't throw an error, though:

    const {
      foo,
      bar,
      baz
    } = window

    export default {
      foo,
      bar,
      baz
    }

And the code was working in the original -- only the linter complained about it being undefined, not the JS interpreter.

My understanding is that the original export I pasted should basically be doing:

    export default {
      foo: window.foo,
      bar: window.bar,
      baz: window.baz
    }

and if those values are undefined on window, they should be undefined in my export, which is fine. The only case where this code would fail is if window itself is undefined.

**Comment by sebmck** [comment]
@shipstar because you're defining them. Your original export paste is actually equivalent to:

```javascript
export default window;
```

**Comment by shipstar** [comment]
I think the JS is actually throwing an error as well now that I dig a little deeper, so I think you're right here. Good catch, and sorry for the trouble!
